README.md

```diff
@@ -1,2 +1,99 @@
-# arxiv-fulltext
-arXiv fulltext extraction
+#  arXiv plain text extraction
+
+This service extracts plain text from arXiv PDFs. This is used during the
+submission process as part of routine quality assurance checks. This is also
+used after announcement, to update quality assurance tools (e.g. overlap
+detection) and to make plain text content available to authorized API
+consumers.
+
+## Quick start
+
+The minimal working service cluster requires the API application, the
+worker application, a Docker host (e.g. a dind container), and a Redis for
+the task queue/result backend. The easiest way to get these all running
+together is to use the ``docker-compose.yml`` configuration in the root of
+this repository.
+
+```bash
+docker-compose build    # Will build the API and worker images.
+docker-compose up
+```
+
+This will also start a mock arXiv server that yields a PDF for requests to
+``/pdf/<arxiv id>``. Note that the same PDF is returned no matter what ID is
+requested.
+
+To use the API you will need an auth token with scopes ``fulltext:read`` and
+``fulltext:write``. The easiest way to generate one of these is to use the
+helper script
+[here](https://github.com/arXiv/arxiv-auth/blob/develop/generate_token.py).
+Make sure that you use the same ``JWT_SECRET`` that is used in
+``docker-compose.yml``.
+
+To extract text from announced e-prints, you need only ``fulltext:create``
+and ``fulltext:read``.
+
+To extract text from submissions, you will also need ``compile:read`` and
+``compile:create``. Your user ID will also need to match the owner stated by
+the mock compiler endpoint in [``mock_arxiv.py``](./mock_arxiv.py).
+
+```bash
+JWT_SECRET=foosecret pipenv run python generate_token.py
```

Comment:
`generate_token.py` is currently missing from the repo. Maybe a generic version could be included in the auth or base package?

Reply:
It's in the auth repo: https://github.com/arXiv/arxiv-auth/blob/develop/generate_token.py

Reply:
My bad! I promise I RTM, but it seems a little bit too quickly :-) 

Still, it would be nice to be able to call such scripts from the same environment, which already installs several arxiv packages. For example, [this](https://github.com/arXiv/arxiv-auth/blob/develop/users/arxiv/users/helpers.py) is in `arxiv-auth` and similarly has a `generate_token` function; the package setup could for example include a client script like `generate_token.py` that leverages that helper function. I think it's fine for this to be out of scope for this PR though.

Reply:
> For example, [this](https://github.com/arXiv/arxiv-auth/blob/develop/users/arxiv/users/helpers.py) is in `arxiv-auth` and similarly has a `generate_token` function; the package setup could for example include a client script like `generate_token.py` that leverages that helper function.

Absolutely agree. Something like:

```bash
pipenv run python -m arxiv.auth.generate_token
```

Reply:
Yes, exactly

Reply:
https://arxiv-org.atlassian.net/browse/ARXIVNG-2131